tests/Tests/ORM/Tools/SchemaToolTest.php

```diff
@@ -166,12 +179,61 @@ public function testPostGenerateEvents(): void
             $em->getClassMetadata(CmsUser::class),
         ];
 
-        $schema = $schemaTool->getSchemaFromMetadata($classes);
+        $schemaTool = new SchemaTool($em);
+        $schemaTool->getSchemaFromMetadata($classes);
 
         self::assertEquals(count($classes), $listener->tableCalls);
         self::assertTrue($listener->schemaCalled);
     }
 
+    public function testGetUpdateSchemaSqlEvents(): void
+    {
+        $em = $this->getTestEntityManager();
+
+        // Listen to events.
+        $em->getEventManager()->addEventListener([ToolEvents::postGenerateSchema, ToolEvents::postGenerateComparisonSchema], new class {
+            public function postGenerateSchema(GenerateSchemaEventArgs $eventArgs): void
+            {
+                $eventArgs->getSchema()->createTable('test_new');
+            }
+
+            public function postGenerateComparisonSchema(GenerateSchemaEventArgs $eventArgs): void
+            {
+                $eventArgs->getSchema()->createTable('test_old');
+            }
+        });
+
+        $introspectedSchema = new class () extends Schema {
+            public static bool $called = false;
+
+            /** {@inheritDoc} */
+            public function createTable($name)
```

Comment:
```suggestion
            public function createTable($name): Table
```

Reply:
Note that the return type of `createTable` is `\Doctrine\DBAL\Schema\Table`, as opposed to the existing import of `Doctrine\ORM\Mapping\Table`.